chartfx-dataset/src/main/java/de/gsi/dataset/serializer/spi/BinarySerialiser.java

```diff
@@ -279,383 +313,397 @@ public static double getDouble(final IoBuffer readBuffer) {
+    public <E> List<E> getList(final List<E> collection, BiFunction<Type, Type[], FieldSerialiser<E>> serialiserLookup) {
+        buffer.getArraySizeDescriptor();
+        final int nElements = buffer.getInt();
+        final DataType listDataType = getDataType(buffer.getByte());
+        final DataType valueDataType = getDataType(buffer.getByte());
+        if (!listDataType.equals(DataType.LIST) && !listDataType.equals(DataType.COLLECTION)) {
+            throw new IllegalArgumentException("dataType incompatible with List = " + listDataType);
+        }
+        final List<E> retCollection;
+        if (collection == null) {
+            retCollection = new ArrayList<>();
+        } else {
+            retCollection = collection;
+            retCollection.clear();
+        }
+
+        if (DataType.OTHER.equals(valueDataType)) {
+            final String classTypeName = buffer.getStringISO8859();
+            final String secondaryTypeName = buffer.getStringISO8859();
+            final Type classType = ClassUtils.getClassByName(classTypeName);
+            final Type[] secondaryType = secondaryTypeName.isEmpty() ? new Type[0] : new Type[] { ClassUtils.getClassByName(secondaryTypeName) };
+            if (serialiserLookup == null) {
+                throw new IllegalArgumentException("protocol error: serialiserLookup must not be null for DataType == OTHER");
+            }
+            final FieldSerialiser<E> serialiser = serialiserLookup.apply(classType, secondaryType);
 
-        return retCollection;
-    }
+            if (serialiser == null) {
+                throw new IllegalArgumentException(NO_SERIALISER_IMP_FOUND + classTypeName);
+            }
+            for (int i = 0; i < nElements; i++) {
+                retCollection.add(serialiser.getReturnObjectFunction().apply(this, null, null));
+            }
 
-    public static long getLong(final IoBuffer readBuffer) {
-        if (readBuffer.hasRemaining()) {
-            return readBuffer.getLong();
+            return retCollection;
         }
-        throw new IndexOutOfBoundsException(READ_POSITION_AT_BUFFER_END);
-    }
 
-    public static long[] getLongArray(final IoBuffer readBuffer) {
-        if (readBuffer.hasRemaining()) {
-            return readBuffer.getLongArray();
+        // read primitive or String value vector
+        final E[] values = getGenericArrayAsBoxedPrimitive(valueDataType);
+        if (nElements != values.length) {
+            throw new IllegalStateException(PROTOCOL_MISMATCH_N_ELEMENTS_HEADER + nElements + VS_ARRAY + values.length);
         }
-        throw new IndexOutOfBoundsException(READ_POSITION_AT_BUFFER_END);
+        retCollection.addAll(Arrays.asList(values));
+
+        return retCollection;
     }
 
-    public static <K, V> Map<K, V> getMap(final IoBuffer readBuffer, final Map<K, V> map) {
+    @Override
+    public <K, V, E> Map<K, V> getMap(final Map<K, V> map, BiFunction<Type, Type[], FieldSerialiser<E>> serialiserLookup) {
```

Comment:
Codacy found an issue: [The method 'getMap(Map, BiFunction)' has an NPath complexity of 648, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5905529)